pykilosort/gui/main.py

```diff
@@ -0,0 +1,216 @@
+            elif event.key() == QtCore.Qt.Key_C:
+                self.toggle_view()
+            elif event.key() == QtCore.Qt.Key_1:
+                self.toggle_mode("raw")
+            elif event.key() == QtCore.Qt.Key_2:
+                self.toggle_mode("whitened")
+            elif event.key() == QtCore.Qt.Key_3:
+                self.toggle_mode("prediction")
+            elif event.key() == QtCore.Qt.Key_4:
+                self.toggle_mode("residual")
+            else:
+                pass
+            event.accept()
+        else:
+            event.ignore()
+
+    def setup(self):
+        self.setWindowTitle(f"Kilosort{__version__}")
+
+        self.content_layout.addWidget(self.header_box, 3)
+
+        self.second_boxes_layout.addWidget(self.settings_box, 85)
+        self.second_boxes_layout.addWidget(self.run_box, 15)
+
+        self.boxes_layout.addLayout(self.second_boxes_layout, 20)
+        self.boxes_layout.addWidget(self.probe_view_box, 10)
+        self.boxes_layout.addWidget(self.data_view_box, 70)
+
+        self.boxes.setLayout(self.boxes_layout)
+        self.content_layout.addWidget(self.boxes, 90)
+        self.content_layout.addWidget(self.message_log_box, 7)
+
+        self.content_layout.setContentsMargins(10, 10, 10, 10)
+        self.content.setLayout(self.content_layout)
+        self.setCentralWidget(self.content)
+
+        self.settings_box.settingsUpdated.connect(self.set_parameters)
+
+        self.data_view_box.channelChanged.connect(self.probe_view_box.update_probe_view)
+        self.data_view_box.modeChanged.connect(self.probe_view_box.synchronize_data_view_mode)
+
+        self.probe_view_box.channelSelected.connect(self.data_view_box.change_primary_channel)
+
+        self.run_box.updateContext.connect(self.update_context)
+
+    def change_channel(self, shift):
+        if self.context is not None:
+            self.data_view_box.shift_primary_channel(shift)
+
+    def shift_data(self, time_shift):
+        if self.context is not None:
+            self.data_view_box.shift_current_time(time_shift)
+
+    def change_displayed_channel_count(self, shift):
+        if self.context is not None:
+            self.data_view_box.change_displayed_channel_count(shift)
+
+    def toggle_view(self):
+        self.data_view_box.traces_button.toggle()
+
+    def toggle_mode(self, mode):
+        if mode == "raw":
+            self.data_view_box.raw_button.toggle()
+        elif mode == "whitened":
+            self.data_view_box.whitened_button.toggle()
+        elif mode == "prediction":
+            self.data_view_box.prediction_button.toggle()
+        elif mode == "residual":
+            self.data_view_box.residual_button.toggle()
+        else:
+            raise ValueError("Invalid mode requested!")
+
+    def set_parameters(self):
+        settings = self.settings_box.settings
+        advanced_options = self.settings_box.advanced_options
+
+        self.data_path = settings.pop('data_file_path')
```

Comment:
why do we remove these things from the settings when accessing them?

i.e. why not `get`?

Reply:
(if there is a reason just add as a comment I think :) )

Reply:
The parameters I am "popping" are not a part of the default KilosortParams class (nor were they a part of the former Params dictionary), so I remove them from the settings object before parsing the other settings values. In addition, I need the "popped" parameters separately in multiple other scenarios, so I store them as instance variables.